test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_kernel.cpp

```diff
@@ -0,0 +1,1453 @@
+            nullptr /* arg_svm_list - nullptr means no change*/,
+            nullptr /* exec_info_list */,
+            nullptr /* global_work_offset */,
+            nullptr /* global_work_size */,
+            nullptr /* local_work_size */
+        };
+        cl_mutable_base_config_khr mutable_config{
+            CL_STRUCTURE_TYPE_MUTABLE_BASE_CONFIG_KHR, nullptr, 1,
+            &dispatch_config
+        };
+        error =
+            clUpdateMutableCommandsKHR(mutable_command_buffer, &mutable_config);
+        test_error(error, "clUpdateMutableCommandsKHR failed");
+
+        error =
+            clEnqueueReadBuffer(queue, streams[1], CL_TRUE, 0, sizeToAllocate,
+                                resultData.data(), 0, NULL, NULL);
+        test_error(error, "clEnqueueReadBuffer failed");
+
+        for (int i = 0; i < (int)(numberOfInts); i++)
+            if (constantData[i] != resultData[i])
+            {
+                log_error("Data failed to verify: constantData[%d]=%d != "
+                          "resultData[%d]=%d\n",
+                          i, constantData[i], i, resultData[i]);
+                return TEST_FAIL;
+            }
+
+        return TEST_PASS;
+    }
+
+    clUpdateMutableCommandsKHR_fn clUpdateMutableCommandsKHR = nullptr;
+    clCommandBufferWrapper mutable_command_buffer;
+    cl_mutable_command_khr command = nullptr;
+
+    const cl_ulong max_size = 16;
+    cl_ulong currentSize;
+};
+
+struct MutableDispatchPODArguments : public BasicMutableCommandBufferTest
+{
+    using BasicMutableCommandBufferTest::BasicMutableCommandBufferTest;
+
+    MutableDispatchPODArguments(cl_device_id device, cl_context context,
+                                cl_command_queue queue)
+        : BasicMutableCommandBufferTest(device, context, queue),
+          mutable_command_buffer(this)
+    {}
+
+    virtual cl_int SetUp(int elements) override
+    {
+        BasicMutableCommandBufferTest::SetUp(elements);
+
+        cl_platform_id platform;
+        cl_int error =
+            clGetDeviceInfo(device, CL_DEVICE_PLATFORM, sizeof(cl_platform_id),
+                            &platform, nullptr);
+        test_error(error, "clGetDeviceInfo for CL_DEVICE_PLATFORM failed");
+
+        GET_EXTENSION_ADDRESS(clUpdateMutableCommandsKHR);
+
+        cl_command_buffer_properties_khr properties[3] = {
+            CL_COMMAND_BUFFER_FLAGS_KHR, 0, 0
+        };
+        properties[1] = CL_COMMAND_BUFFER_MUTABLE_KHR;
+
+        mutable_command_buffer =
+            clCreateCommandBufferKHR(1, &queue, properties, &error);
+        test_error(error, "clCreateCommandBufferKHR failed");
+
+        return 0;
+    }
+
+    cl_int Run() override
+    {
+        const char *sample_const_arg_kernel[] = {
+            "__kernel void sample_test(__constant int *src, int "
+            "dst )\n"
+            "{\n"
+            "    size_t  tid = get_global_id(0);\n"
+            "\n"
+            "    dst = src[tid];\n"
```

Comment:
can we make `dst` a buffer instead and `src` a scalar `int` so that we can verify the results.